[INFO   ] Logger: Record log in /storage/emulated/0/code/util/DownWithYT/v1.1/.kivy/logs/kivy_22-09-19_16.txt
[INFO   ] Kivy: v2.0.0
[INFO   ] Kivy: Installed at "/data/user/0/ru.iiec.pydroid3/files/arm-linux-androideabi/lib/python3.9/site-packages/kivy/__init__.py"
[INFO   ] Python: v3.9.7 (default, Oct  9 2021, 08:02:40) 
[GCC 11.1.0]
[INFO   ] Python: Interpreter at "/data/user/0/ru.iiec.pydroid3/files/arm-linux-androideabi/bin/python3"
[INFO   ] Factory: 186 symbols loaded
[INFO   ] Image: Providers: img_tex, img_dds, img_sdl2, img_pil (img_ffpyplayer ignored)
[INFO   ] Text: Provider: sdl2
[INFO   ] Window: Provider: sdl2
[INFO   ] GL: Using the "OpenGL ES 2" graphics system
[INFO   ] GL: Backend used <sdl2>
[INFO   ] GL: OpenGL version <b'OpenGL ES 3.2 build 1.9@4893595'>
[INFO   ] GL: OpenGL vendor <b'Imagination Technologies'>
[INFO   ] GL: OpenGL renderer <b'PowerVR Rogue GE8300'>
[INFO   ] GL: OpenGL parsed version: 3, 2
[INFO   ] GL: Texture max size <4096>
[INFO   ] GL: Texture max units <16>
[INFO   ] Window: auto add sdl2 input provider
[INFO   ] Window: virtual keyboard not allowed, single mode, not docked
[INFO   ] GL: NPOT texture support is available
[WARNING] stderr: Traceback (most recent call last):
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/arm-linux-androideabi/lib/python3.9/site-packages/kivy/lang/builder.py", line 242, in create_handler
[WARNING] stderr:     return eval(value, idmap), bound_list
[WARNING] stderr:   File "/storage/emulated/0/code/util/DownWithYT/v1.1/dwyt.kv", line 23, in <module>
[WARNING] stderr:     center_x:root.parent.center_x
[WARNING] stderr: AttributeError: 'NoneType' object has no attribute 'center_x'
[WARNING] stderr: 
[WARNING] stderr: During handling of the above exception, another exception occurred:
[WARNING] stderr: 
[WARNING] stderr: Traceback (most recent call last):
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/arm-linux-androideabi/lib/python3.9/site-packages/kivy/lang/builder.py", line 695, in _apply_rule
[WARNING] stderr:     value, bound = create_handler(
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/arm-linux-androideabi/lib/python3.9/site-packages/kivy/lang/builder.py", line 245, in create_handler
[WARNING] stderr:     raise BuilderException(rule.ctx, rule.line,
[WARNING] stderr: kivy.lang.builder.BuilderException: Parser: File "/storage/emulated/0/code/util/DownWithYT/v1.1/dwyt.kv", line 23:
[WARNING] stderr: ...
[WARNING] stderr:      21:            source:"down_logo.png"
[WARNING] stderr:      22:            size: 400, 400 / self.image_ratio
[WARNING] stderr: >>   23:            center_x:root.parent.center_x
[WARNING] stderr:      24:            center_y:root.parent.center_y
[WARNING] stderr: ...
[WARNING] stderr: AttributeError: 'NoneType' object has no attribute 'center_x'
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/arm-linux-androideabi/lib/python3.9/site-packages/kivy/lang/builder.py", line 242, in create_handler
[WARNING] stderr:     return eval(value, idmap), bound_list
[WARNING] stderr:   File "/storage/emulated/0/code/util/DownWithYT/v1.1/dwyt.kv", line 23, in <module>
[WARNING] stderr:     center_x:root.parent.center_x
[WARNING] stderr: 
[WARNING] stderr: 
[WARNING] stderr: During handling of the above exception, another exception occurred:
[WARNING] stderr: 
[WARNING] stderr: Traceback (most recent call last):
[WARNING] stderr:   File "/storage/emulated/0/code/util/DownWithYT/v1.1/dwyt_app.py", line 40, in <module>
[WARNING] stderr:     DWYTApp().run()
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/arm-linux-androideabi/lib/python3.9/site-packages/kivy/app.py", line 949, in run
[WARNING] stderr:     self._run_prepare()
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/arm-linux-androideabi/lib/python3.9/site-packages/kivy/app.py", line 919, in _run_prepare
[WARNING] stderr:     root = self.build()
[WARNING] stderr:   File "/storage/emulated/0/code/util/DownWithYT/v1.1/dwyt_app.py", line 30, in build
[WARNING] stderr:     self.home_screen = HomeScreen()
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/arm-linux-androideabi/lib/python3.9/site-packages/kivy/uix/floatlayout.py", line 65, in __init__
[WARNING] stderr:     super(FloatLayout, self).__init__(**kwargs)
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/arm-linux-androideabi/lib/python3.9/site-packages/kivy/uix/layout.py", line 76, in __init__
[WARNING] stderr:     super(Layout, self).__init__(**kwargs)
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/arm-linux-androideabi/lib/python3.9/site-packages/kivy/uix/widget.py", line 359, in __init__
[WARNING] stderr:     self.apply_class_lang_rules(
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/arm-linux-androideabi/lib/python3.9/site-packages/kivy/uix/widget.py", line 463, in apply_class_lang_rules
[WARNING] stderr:     Builder.apply(
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/arm-linux-androideabi/lib/python3.9/site-packages/kivy/lang/builder.py", line 541, in apply
[WARNING] stderr:     self._apply_rule(
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/arm-linux-androideabi/lib/python3.9/site-packages/kivy/lang/builder.py", line 710, in _apply_rule
[WARNING] stderr:     raise BuilderException(rule.ctx, rule.line,
[WARNING] stderr: kivy.lang.builder.BuilderException: Parser: File "/storage/emulated/0/code/util/DownWithYT/v1.1/dwyt.kv", line 23:
[WARNING] stderr: ...
[WARNING] stderr:      21:            source:"down_logo.png"
[WARNING] stderr:      22:            size: 400, 400 / self.image_ratio
[WARNING] stderr: >>   23:            center_x:root.parent.center_x
[WARNING] stderr:      24:            center_y:root.parent.center_y
[WARNING] stderr: ...
[WARNING] stderr: BuilderException: Parser: File "/storage/emulated/0/code/util/DownWithYT/v1.1/dwyt.kv", line 23:
[WARNING] stderr: ...
[WARNING] stderr:      21:            source:"down_logo.png"
[WARNING] stderr:      22:            size: 400, 400 / self.image_ratio
[WARNING] stderr: >>   23:            center_x:root.parent.center_x
[WARNING] stderr:      24:            center_y:root.parent.center_y
[WARNING] stderr: ...
[WARNING] stderr: AttributeError: 'NoneType' object has no attribute 'center_x'
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/arm-linux-androideabi/lib/python3.9/site-packages/kivy/lang/builder.py", line 242, in create_handler
[WARNING] stderr:     return eval(value, idmap), bound_list
[WARNING] stderr:   File "/storage/emulated/0/code/util/DownWithYT/v1.1/dwyt.kv", line 23, in <module>
[WARNING] stderr:     center_x:root.parent.center_x
[WARNING] stderr: 
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/arm-linux-androideabi/lib/python3.9/site-packages/kivy/lang/builder.py", line 695, in _apply_rule
[WARNING] stderr:     value, bound = create_handler(
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/arm-linux-androideabi/lib/python3.9/site-packages/kivy/lang/builder.py", line 245, in create_handler
[WARNING] stderr:     raise BuilderException(rule.ctx, rule.line,
[WARNING] stderr: 
